cfpb/cfgov-refresh
# Switch from gulp-changed to gulp-newer in style and script tasks

This will enable us to check more source files than what is being passed through the pipe, fixing the bug where assets aren't rebuilt if required/imported JS/CSS is updated.


## Removals

- Unused `lib` config


## Changes

- Replaces gulp-changed step with gulp-newer in style and script tasks
- Updates cf-atomic-component and cf-expandable dependencies


## Testing

1. Pull branch
1. `npm install`
1. `gulp clean`

### Testing style tasks

1. Run `gulp styles:modern` to establish a baseline time
1. Make a trivial edit to a file that is imported into `main.less`, such as `breadcrumbs.less`
1. Run `gulp styles:modern` and see a similar time as before
1. Immediately re-run `gulp styles:modern` and see that it finishes in under 100ms!
1. Repeat similarly with other subtasks, as desired

### Testing script tasks

1. Run `gulp scripts:modern` to establish a baseline time
1. Make a trivial edit to a file that is require into `common.js`, such as `modules/focus-target.js`
1. Run `gulp scripts:modern` and see a similar time as before
1. Immediately re-run `gulp scripts:modern` and see that it finishes in under 100ms!
1. Repeat similarly with other subtasks, as desired


## Notes

- By default, gulp-newer will compare the destination file's modified time against:
  - All the CSS and JS in our `cfgov/unprocessed/` folder
  - The `node_modules/` folder (whose timestamp gets updated any time a dependency is updated)
  - JS files in `config/` and `gulp/`

## Todos

- Migrate other gulp-changed usage (in `copy` and `images` scripts) to gulp-newer?

## Checklist
